kotlin/raster-function-service/README.md

```diff
@@ -1,54 +1,43 @@
-# Raster Function Service
+# Raster function service
 
-Raster functions are operations performed on a `Raster`to apply on-the-fly processing such as Hillshade.  This sample demonstrates how to create an `ImageServiceRaster` and apply a `RasterFunction` to it. The `RasterFunction` applied in this case is Hillshade.
+Load a raster from a service, then apply a function to it.
 
-![Raster Function App](raster-function-service.png)
+![Image of apply raster function to raster from service](raster-function-service.png)
 
-## Features
+## Use case
+
+Raster functions allow processing operations that can be applied to one or more rasters on the fly. Functions can be applied to rasters that come from a service. A land survey agency may apply hillshade and aspect functions to rasters with elevation data in order to better determine the topography of a landscape and to make further planning decisions.
+
+## How to use the sample
+
+The raster function is applied automatically when the sample starts and the result is displayed.
+
+## How it works
+
+1. Create the `ImageServiceRaster` referring to the image server URL.
+2. Create the `RasterFunction` from a JSON string.
+3. Get the name of the raster argument to the function with `rasterFunction.arguments.rasterNames[0]`.
+4. Set the raster argument with `rasterFunction.setRaster(rasterName, imageServiceRaster)`.
```

Comment:
Looking at the code it seems like this is a bit different from common; lines 80-90

```
// create raster function from json string
    RasterFunction rasterFuntionFromJson = RasterFunction.fromJson(getString(R.string.hillshade_simplified));
    // get parameter name value pairs used by hillside
    RasterFunctionArguments rasterFunctionArguments = rasterFuntionFromJson.getArguments();
    // get a list of raster names associated with the raster function
    List<String> rasterNames = rasterFunctionArguments.getRasterNames();
    rasterFunctionArguments.setRaster(rasterNames.get(0), raster);
    // create raster as raster layer
    raster = new Raster(rasterFuntionFromJson);
    RasterLayer hillshadeLayer = new RasterLayer(raster);
    mMapView.getMap().getOperationalLayers().add(hillshadeLayer);
```

I would suggest:

```
3. Get the arguments of the raster function with RasterFunction.getArguments()
4. Get the names of the rasters with RasterFunctionArguments.getRasterNames()
5. Set the raster argument with `RasterFunctionArguments.setRaster(rasterName, imageServiceRaster)
```

But maybe there's a way to shorten that back down to two steps.

Reply:
Sorry, didn't realize this was kotlin at first, so I suppose this would be the same but with property access

Reply:
the main steps are captured above, but the common design has bunched two steps together. I'll separate out the getters of arguments and rasternames to be more readable.